community/modules/compute/schedmd-slurm-gcp-v5-node-group/source_image_logic.tf

```diff
@@ -16,52 +16,46 @@
+    "projects/${data.google_compute_image.slurm.project}/global/images"
   )
+  source_image_family = can(var.instance_image.family) ? data.google_compute_image.slurm.family : ""
+  source_image        = can(var.instance_image.name) ? data.google_compute_image.slurm.name : ""
 }
 
 data "google_compute_image" "slurm" {
-  family  = contains(keys(var.instance_image), "family") ? var.instance_image["family"] : null
-  name    = contains(keys(var.instance_image), "name") ? var.instance_image["name"] : null
-  project = local.source_image_project
+  family  = try(var.instance_image.family, null)
+  name    = try(var.instance_image.name, null)
+  project = var.instance_image.project
 
   lifecycle {
     postcondition {
-      condition = !var.instance_image_compatible || try(contains(local.supported_project_families[self.project], self.family), false)
-      error_message = (contains(keys(local.supported_project_families), self.project) ? <<-EOD
-      The image specified in the blueprint was found, but it is not compatible with this HPC Toolkit for Slurm module. The compatible image families are:
-
-      ${format("%-30s%-30s", "PROJECT", "IMAGE FAMILY")}
-      ${join("\n", [for proj in keys(local.supported_project_families) : join("\n", [for family in local.supported_project_families[proj] : format("%-30s%-30s", proj, family)])])}
-
-      Images that are not within this family may not have Slurm installed by default or may have faulty configurations.  If you are attempting to deploy a custom Slurm image, or an image with an unexpected version, please set the "instance_image_compatible" variable to "false" in the blueprint.
-      See https://github.com/GoogleCloudPlatform/hpc-toolkit/blob/main/docs/vm-images.md#slurm-on-gcp for more information.
+      condition     = var.instance_image_custom || try(contains(keys(local.known_project_families), self.project), false)
+      error_message = <<-EOD
+      '${self.project}' is not a known project with compatible Slurm image families. Use the 'instance_image_custom' flag to deploy custom images. See: https://github.com/GoogleCloudPlatform/hpc-toolkit/blob/main/docs/vm-images.md#slurm-on-gcp.
       EOD
-        : <<-EOD
-      The project specified in "instance_image" was found, but is not guaranteed to contain images that are compatible with this module.  The project/s with compatible images are:
-
-      PROJECT
-      ${join("\n", [for proj in keys(local.supported_project_families) : proj])}
-
-      Please update the project specification in "instance_image", or set the "instance_image_compatible" variable to "false".
-      See https://github.com/GoogleCloudPlatform/hpc-toolkit/blob/main/docs/vm-images.md#slurm-on-gcp for more information.
+    }
+    postcondition {
+      condition     = var.instance_image_custom || !try(contains(keys(local.known_project_families), self.project), false) || try(contains(local.known_project_families[self.project], self.family), false)
```

Comment:
I don't think `var.instance_image_custom` should appear in this condition. The logic is that the first condition will fire if you aren't using the schedmd-slurm-public project (in which case you must declare your intent to use a custom image) and the 2nd condition will fire if you are using schedmd-slurm-public but are using an incompatible family. There is not a situation where we want people to be able to override that logic within schedmd-slurm-public.

Reply:
I think the logic reduces to:

```hcl
!contains(keys(local.known_project_families), self.project) || try(contains(local.known_project_families[self.project], self.family)
```

If Terraform did short-circuiting on booleans, you wouldn't need the `try`. But it doesn't.

Reply:
I'll make the fix removing the first condition, but it also makes it impossible for users to utilize (at their own peril) other schedmd images with the wrapper modules for v5.  I'd argue that `instance_image_custom: true` already lets them use any image, so why not other schedmd images?